libs/s25main/world/GameWorldBase.cpp

```diff
@@ -696,3 +696,14 @@ void GameWorldBase::RecalcBQ(const MapPoint pt)
         GetNotifications().publish(NodeNote(NodeNote::BQ, pt));
     }
 }
+
+void GameWorldBase::SetComputerBarrier(const MapPoint& pt, unsigned radius)
+{
+    for(const auto& pt : GetPointsInRadiusWithCenter(pt, radius))
+        ptsInsideComputerBarriers.insert(pt);
+}
+
+bool GameWorldBase::IsInsideComputerBarrier(const MapPoint& pt) const
+{
+    return ptsInsideComputerBarriers.count(pt);
```

Comment:
Maybe rather use `helpers::contains` which is more readable and doesn't coerce a `std::size_t` to a bool (IIRC MSVC warns about that)

Have you evaluated performance? As this is called quite often when searching for a building spot it likely matters. I imagine a `flat_map` might help or just storing point-radius pairs and comparing the distance. A plain `std::set` might cause a larger hit due to memory accesses and `MapPoint` being small (32 bit) at least a flat_map might be beneficial

Reply:
Replaced count with helpers::contains.

Ran a quick performance check with VS + RelWithDebInfo. This function is irrelevant in the grand scheme of things, with either count or contains:
![Zrzut ekranu 2024-08-05 183218](https://github.com/user-attachments/assets/bb7eb7d2-8eb9-47d3-9568-ec9dc35daf21)
![Zrzut ekranu 2024-08-05 182641](https://github.com/user-attachments/assets/02f88d21-fd5e-4aca-a8d2-554c1f0a8e1d)
![Zrzut ekranu 2024-08-05 182624](https://github.com/user-attachments/assets/f3c8c808-d4e1-4fc2-a3a5-710fdb357321)

Reply:
> This function is irrelevant in the grand scheme of things, with either count or contains:

I was more thinking about the "death by a thousand cuts": If we don't at least think about this for anything with loops the sum of those functions will be slow.

There might be a flaw in your performance chart: It looks like those counts are "inclusive times", i.e. `GameManager.Run` includes the time of e.g. `GameClient.Run` and `GameServer.Run` so the percentages don't add up to 100% which leads to those many 0.01% times. Especially as the network and draw calls are included. I guess including only `AIPlayerJH.Run` and below is more expressive.

What is the highest number of barriers and how many points does that yield? And what is "typical"? Maybe it really is irrelevant...